src/io/fileops.c

```diff
@@ -290,7 +290,17 @@ static int are_we_group_member(MVMThreadContext *tc, gid_t group) {
     }
 FILE_IS(readable, R)
 FILE_IS(writable, W)
-FILE_IS(executable, X)
+    MVMint64 MVM_file_isexecutable (MVMThreadContext *tc, MVMString *filename, MVMint32 use_lstat) {
+        if (!MVM_file_exists(tc, filename, use_lstat))
+            return 0;
+        else {
+            uv_stat_t statbuf = file_info(tc, filename, use_lstat);
+            MVMint64 r = (statbuf.st_mode & S_IXOTH)
```

Comment:
Hm, can we not determine whether the file exists using the result in `uv_stat_t`? That'd seem to avoid two system calls, plus it's a race between the exists check and asking for the file info so it can't be relied upon anyway.

Reply:
I have refactored the file respectively.